### PR TITLE
Send delegate messages for all user-initiated dismissals

### LINKED ISSUE #134: Dismiss bug created in commit c0c2040
By swapping the logic to fix the previous dismissal bug to make interactive dismissals properly call the delegate methods in c0c2040, it looks like it created the same bug for dismissing via the X button in the top left of NYTPhotosViewController. Now, when I swipe to dismiss, the delegate DidDismiss or WillDismiss are called, but not when I click that button.

### PR DESCRIPTION
**Pull request by cdzombak**
- Refactor how dismissals are managed, to allow disambiguating user-initiated and programmatic dismissals for determination of whether to send delegate messages, without tracking an additional bit of state.
- https://github.com/NYTimes/NYTPhotoViewer/commit/16bd3c040b69bd5b2b125d1b8f15d1bb03e272ba should've been covered in #128 , but I just now noticed it
- Other commits in this PR are documentation improvements around this behavior, and minor style improvements.

Closes #134.

### CODE REVIEW

File: Pod/Classes/ios/NYTPhotosViewController.m [hunk 1]
```diff
@@ -53,6 +53,13 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder NS_DESIGNATED_INITIALIZER;
 @property (nonatomic, readonly) UIView *referenceViewForCurrentPhoto;
 @property (nonatomic, readonly) CGPoint boundsCenterPoint;
 
+/**
+ *  Tracks whether the current dismissal of this view controller is due to user interaction
+ *  (an interactive pan or a tap on the Close button).
+ *  Reset to `NO` when an interactive dismissal is not in progress.
+ */
+@property (nonatomic, getter=isCurrentDismissalUserTriggered) BOOL currentDismissalUserTriggered;
```
Comment (bcapps) [inline Pod/Classes/ios/NYTPhotosViewController.m]: Could you potentially eliminate this intermediate state if both `doneButtonTapped:` and `didPanWithGestureRecognizer` called through to a common function, hypothetically something like `dismissViewControllerAnimated:isUserInitiated:completion:`, passing `YES` for `isUserInitiated`, while `dismissViewControllerAnimated:completion:` calls through to the common method with `NO`?

**Comment by soundsokay** [comment]
:+1: save for the (nice to have) expanded unit test coverage noted in the comment in https://github.com/NYTimes/NYTPhotoViewer/commit/eff4f0834a2e05804325fb41f8f7e415664da2db

File: Pod/Classes/ios/NYTPhotosViewController.m (hunk 1 above)
Comment (cdzombak) [inline Pod/Classes/ios/NYTPhotosViewController.m]: @bcapps :100:

**Comment by cdzombak** [comment]
screen shot 2016-01-26 at 10 43 07

Currently this test is failing, though setting a breakpoint in `-dismissViewControllerAnimated:userInitiated:completion:` tells me that that method ​_is_​ being called, with those values (`YES`, `NO`, `nil`). So…what am I missing?

**Comment by bcapps** [comment]
Although the code looks right, the error message seems to imply that it's expecting (`YES`,`YES`,`nil`) instead of what you intended.

**Comment by cdzombak** [comment]
> Although the code looks right, the error message seems to imply that it's expecting (`YES`,`YES`,`nil`) instead of what you intended.

Hm, I must've skimmed over that. I've been looking at OCMock errors too long this morning.

…blowing away `DerivedData` doesn't fix this, so I'm at a bit of a loss.

**Comment by cdzombak** [comment]
https://github.com/NYTimes/NYTPhotoViewer/commit/922bdb0d5bab53221f5d13fe36fd46f162b04b4a makes no difference.

**Comment by soundsokay** [comment]
:+1:

**Pull request merged**